LiteCore/Database/DatabaseImpl.cc

```diff
@@ -658,19 +670,25 @@ namespace litecore {
         // as happens if you call `deleteCollection(coll->spec())`:
         string keyStoreName = collectionNameToKeyStoreName(spec);
         bool isDefault = isDefaultCollection(spec);
-
         Transaction t(this);
 
+        _dataFile->forOtherDataFiles([&keyStoreName](DataFile* df) {
```

Comment:
Strictly speaking, you shouldn't notify until after the `commit()` (or maybe just before?), because the transaction might yet fail if `deleteKeyStore` fails.

Reply:
I see the logic there.  I think the benefit is enough here because the worst that happens if the call fails is that the next time a collection on this table is used, it will do a no-op create (CREATE IF NOT EXISTS).  Otherwise you have a situation where the table is gone, but the other database instances are not informed yet.  A creation during this window would cause issues with nonexistent tables.

Reply:
Interesting I had similar train of thoughts. In my abandoned PR, I actually tried to set the (existence) flags only when I know when it's going to commit. As I was reviewing this PR, I was really thinking it and reasoned same path. If the flag is mistakenly nonexistent, it's not fatal because, as Jim pointed out, we create new table with CREATE IF NOT EXISTS. The reverse error is detrimental: the flag says committed, but table actually missing.